ThriftHandler/MapDHandler.cpp

```diff
@@ -641,6 +641,62 @@ void MapDHandler::sql_execute(TQueryResult& _return,
   }
 }
 
+void MapDHandler::sql_execute_df(TGpuDataFrame& _return,
+                                 const TSessionId& session,
+                                 const std::string& query_str,
+                                 const int32_t first_n) {
+#ifdef ENABLE_ARROW_CONVERTER
+  const auto session_info = MapDHandler::get_session(session);
+  int64_t execution_time_ms = 0;
+  LOG(INFO) << query_str;
+  int64_t total_time_ms = measure<>::execution([&]() {
+    SQLParser parser;
+    std::list<std::unique_ptr<Parser::Stmt>> parse_trees;
+    std::string last_parsed;
+#ifdef HAVE_CALCITE
```

Comment:
It's safe to assume that `HAVE_CALCITE` and `HAVE_RAVM` are always defined at this point. We should find some time to remove the legacy clutter, sorry about it.